### PR TITLE
Rebuild for libboost 1.82

### PR DESCRIPTION
**Pull request by h-vetinari**
Bot failed, likely due to cling requiring a too-old libcxx on osx.

### CODE REVIEW

**Comment by saraedum** [comment]
This needs https://github.com/conda-forge/cpycppyy-feedstock/pull/43.

**Comment by h-vetinari** [comment]
We need to update the `cpycppyy` in `cppyy` (not sure why it's pinned so [tightly](https://github.com/conda-forge/cppyy-feedstock/blob/main/recipe/meta.yaml#L41-L42) there).

**Comment by saraedum** [comment]
The pinning is so tight because upstream does the same. The four packages are basically considered a unit by upstream.

But since cppyy-backend broke that pinning already, there is no point to keep it right now. cpycppyy latest version has no changes, it just has different pins, so there is no harm in upgrading here.

**Comment by saraedum** [comment]
So, effectively this needs https://github.com/conda-forge/cppyy-cling-feedstock/pull/57, https://github.com/conda-forge/cpycppyy-feedstock/pull/44, and a pull request for cppyy to build for python 3.11.

**Comment by h-vetinari** [comment]
> The pinning is so tight because upstream does the same. The four packages are basically considered a unit by upstream.

Which is why I hadn't merged https://github.com/conda-forge/cpycppyy-feedstock/pull/43 on my own... ;-)

**Comment by h-vetinari** [comment]
> So, effectively this needs https://github.com/conda-forge/cppyy-cling-feedstock/pull/57, https://github.com/conda-forge/cpycppyy-feedstock/pull/44, and a pull request for cppyy to build for python 3.11.

I think you might have mixed something up there w.r.t. py311 vs. py312. We shouldn't need anything for py312 at the moment (i.e. the first PR you reference; cppyy-cling already has py311 builds), but we do need the latter two things you mention.

File: .ci_support/osx_64_.yaml [hunk 1]
```diff
@@ -16,18 +14,19 @@ cxx_compiler_version:
...
 - x86_64-apple-darwin13.4.0
 pin_run_as_build:
-  boost-cpp:
-    max_pin: x.x.x
   python:
     min_pin: x.x
     max_pin: x.x
 python:
 - 3.10.* *_cpython
+- 3.11.* *_cpython
```
Comment (saraedum) [inline .ci_support/osx_64_.yaml:29]: Strangely, there is no Python 3.11 build of cppyy. It seems that the Python 3.11 migration never made it to cppyy, even though its dependencies seem to have Python 3.11 builds.

**Comment by saraedum** [comment]
> > So, effectively this needs [conda-forge/cppyy-cling-feedstock#57](https://github.com/conda-forge/cppyy-cling-feedstock/pull/57), [conda-forge/cpycppyy-feedstock#44](https://github.com/conda-forge/cpycppyy-feedstock/pull/44), and a pull request for cppyy to build for python 3.11.
> 
> I think you might have mixed something up there w.r.t. py311 vs. py312. We shouldn't need anything for py312 at the moment (i.e. the first PR you reference; cppyy-cling already has py311 builds), but we do need the latter two things you mention.

Yes, I got confused here. So, somehow the Python 3.11 migration didn't make it to cppyy. I guess if I get the Python 3.12 migration to pass on cppyy-cling, then eventually, I get the Python 3.12 migration to run on cppyy which is also giving us a Python 3.11 cppyy.

**Comment by h-vetinari** [comment]
Hopefully last puzzle piece: https://github.com/conda-forge/cppyy-feedstock/pull/69

**Pull request merged**